BaragonServiceIntegrationTests/src/test/java/com/hubspot/baragon/BaragonServiceTestIT.java

```diff
@@ -0,0 +1,478 @@
+        System.out.println("Starting: " + description.getMethodName());
+     }
+     protected void succeeded(Description description) {
+       System.out.println("\u001B[32mTest passed\u001B[0m");
+     }
+     protected void failed(Throwable e, Description description) {
+       System.out.println("\u001B[31mTest failed\u001B[0m");
+     }
+  };
+  
+  
+  @Rule
+  public Stopwatch stopwatch = new Stopwatch() {
+      @Override
+      protected void finished(long nanos, Description description) {
+        System.out.println("(" + nanos / 1000000000.0 + " s)");
+      }
+  };
+  
+  @After
+  public void teardown() {
+    baragonServiceClient = null;
+  }
+
+  // ----------------------------------------------------
+  
+  @Test
+  public void testStatus() throws Exception {
+    Optional<BaragonServiceStatus> status = baragonServiceClient.getAnyBaragonServiceStatus();
+    assertTrue(status.get().isLeader());
+  }
+
+  @Test
+  public void testWorkers() throws Exception {
+    assertFalse(baragonServiceClient.getBaragonServiceWorkers().isEmpty());
+  }
+  
+  @Test
+  public void testLoadBalancers() throws Exception {
+    assertFalse(baragonServiceClient.getLoadBalancerGroups().isEmpty());
+  }
+  
+  @Test
+  public void testClusterAgents() throws Exception {
+    BaragonAgentMetadata metadata = baragonServiceClient.getLoadBalancerGroupAgentMetadata(LOAD_BALANCER_GROUP).toArray(new BaragonAgentMetadata[0])[0];
+    assertTrue(metadata.getBaseAgentUri() != null && !metadata.getBaseAgentUri().isEmpty());
+  }
+  
+  @Test
+  public void testClusterKnownAgents() throws Exception {
+    BaragonAgentMetadata metadata = baragonServiceClient.getLoadBalancerGroupKnownAgentMetadata(LOAD_BALANCER_GROUP).toArray(new BaragonAgentMetadata[0])[0];
+    assertTrue(metadata.getBaseAgentUri() != null && !metadata.getBaseAgentUri().isEmpty());
+    
+    testDeleteKnownAgent();
+  }
+  
+  // ----------------------------------------------------
+  
+  @Test
+  public void testValidRequest() throws Exception {
+    String serviceId = setupTestService();
+    while(baragonServiceClient.getRequest(serviceId).get().getLoadBalancerState().equals(BaragonRequestState.WAITING)) { }
+    
+    if(baragonServiceClient.getRequest(serviceId).get().getLoadBalancerState().equals(BaragonRequestState.SUCCESS)) {
+      testGetService();
+      testState();
+      testClusterBasePaths();
+      
+      removeService(serviceId);
+    } 
+    else {
+      BaragonResponse resp = baragonServiceClient.getRequest(serviceId).get();
+      throw new Exception("Request did not succeed: " + resp.getMessage().get().toString() 
+          + "\nState: " + resp.getLoadBalancerState());
+    }
+  }
+  
+  public void testClusterBasePaths() {
+    String groupName = baragonServiceClient.getLoadBalancerGroups().toArray(new String[0])[0];
+    String[] basePaths = baragonServiceClient.getOccupiedBasePaths(groupName).toArray(new String[0]);
```

Comment:
Performance-wise it isn't great practice to call toArray with an argument that has length zero/less that size of the collection. It then has to recalculate the size of the array it is creating. That said, we can probably get away without making this an array and just use the collection returned by the client. The `assertTrue(basePaths.length > 0)` can just assert that the collection is not empty, and ` basePaths[0]` on the following line can just call `iterator().next()` like you are using on objects in a few other places